### DISCUSSION TITLE
Initializing error after Samsung system update

### DISCUSSION THREAD

**Discussion by kiselevchik** [NightscoutFoundation/xDrip#3375] [Bug reports] [answered]
Hey,

After a monthly Samsung update xdrip shows initializing errors and I get no readings. 
Would be glad to get any input on that. Phone S24 ultra.

![Screenshot_20240302_173650_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/161962594/5e7cc5fc-193b-4e40-bb6c-b7201d615538)

**Comment by Navid200** [comment]
After an OS update, it is common for some Android settings, like battery optimization, to have been reset to default.  So, make sure xDrip is still excluded. 

If all the [settings](https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html) are correct, go to the Android Bluetooth settings page and delete any devices starting with DXCM.

Then, go to the status page and raise the volume or disable system mute and wait for a pair request and approve when you get it.

**Reply by kiselevchik** [reply]
All checked. Location allowed, battery optimization is off. Bluetooth settings are not available on Samsung anyway. 
After I disconnected the paired DXCM there is no pairing request from xdrip anymore and the same initializing errors keep coming.

**Reply by Navid200** [reply]
How did you disconnect DXCM if there are no Bluetooth settings?

**Reply by kiselevchik** [reply]
I mean "Bluetooth battery optimization" option, as said in Basics.
But I unpaired the sensor, now it's gone from the list of devices and there is no pairing request.

**Reply by kiselevchik** [reply]
Updated to the last nightly version. Still initializing error.

**Reply by Navid200** [reply]
OK,

Is there any other phone with xDrip or Dexcom app trying to connect to the same G7?

Do you happen to have a Dexcom receiver?

**Reply by kiselevchik** [reply]
No other phones. I have a receiver, but haven't used it for months, it's off.

**Reply by Navid200** [reply]
Turn the receiver on and see if it can connect and show readings.

**Reply by kiselevchik** [reply]
Nope. Doesn't see the sensor.

**Reply by Navid200** [reply]
If the receiver does not see the sensor, something is wrong with it.  xDrip cannot be the cause.

**Reply by kiselevchik** [reply]
Sorry, finally found. Is it possible to switch back to the app?

**Reply by Navid200** [reply]
The app the the receiver do not compete.  They can coexist.
Is that what you are asking?

**Reply by kiselevchik** [reply]
I thought G7 has only 1 access point.
But xdrip doesn't get the readings anyway.

**Reply by Navid200** [reply]
No, I believe it has three.

Temporarily change the last character of the transmitter ID in xDrip settings to something else and apply.  Then, change it back.
When you say you get no pairing request, how do you know?  Have you checked the logs?

**Reply by kiselevchik** [reply]
I meant the Bluetooth pairing request that normally pops up with new sender ID. There is none.
Changed the id, xdrip lost connection, changed back to normal and it went directly into initializing errors again.
![Uploading Screenshot_20240302_185409_xDrip+.jpg…]()

**Reply by Navid200** [reply]
And that's exactly why I asked if you have looked at the logs.  You may have a pair request in the logs but because of a phone setting, like very low battery voltage, the phone may not show it.

**Comment by kiselevchik** [comment]
> After an OS update, it is common for some Android settings, like battery optimization, to have been reset to default. So, make sure xDrip is still excluded.
> 
> If all the [settings](https://navid200.github.io/xDrip/docs/G6-Recommended-Settings.html) are correct, go to the Android Bluetooth settings page and delete any devices starting with DXCM.
> 
> Then, go to the status page and raise the volume or disable system mute and wait for a pair request and approve when you get it.

![Screenshot_20240302_190634_xDrip+](https://github.com/NightscoutFoundation/xDrip/assets/161962594/ee039651-1cb9-4ee6-98cc-a7e995b38862)

**Reply by Navid200** [reply, marked as answer]
One brute force way to get it going is:

Backup your database.
Uninstall xDrip.
Remove any device whose name starts with DXCM from Bluetooth paired devices.
Install xDrip and go through the establishing connectivity phase: https://navid200.github.io/xDrip/docs/Dexcom/G7.html

**Reply by kiselevchik** [reply]
Oh man, that's really worked... I was already loosing hope, because concerning the app compatibility S24 is such a pain in the ass even without xdrip. 
Can't say how grateful I am for your help.

**Reply by Navid200** [reply]
Great.

**Reply by rpm5099** [reply]
I had the same problem, the only thing that worked in my scenario was a clean install and NOT reload my database/saved settings.  Complete re-setup from scratch.  Although I was able to reference the saved xml settings while configuring it, doesn't help much for items that you can't translate into gui.  I was never able to figure out what setting was causing the problem.

**Comment by FlyingToto** [comment]
FYI, I had a similar error today, with S25, got scanning errors:2, and I was stuck until I unpaired the DX bluetooth entries and changed the number back and forth in the xdrip config as suggested and it seems to have fixed it. restarting the phone or bluetooth didn't work.

**Reply by Navid200** [reply]
This is a very old thread.  xDrip has changed a lot since then.
There are many comments in this thread by all of us that are now outdated and should not be considered.

These are the troubleshooting steps now:
https://navid200.github.io/xDrip/docs/Dexcom/G7_Troubleshooting.html

**Reply by FlyingToto** [reply]
Indeed but:
1)  interestingly enough this page didn't show in a google search (most likely because Scanning Error:2 is in a picture not in the text)
2) this didn't seem to apply since my sensor had been running fine for the past 8days (it wasn't a new one)... 

maybe we could update the doc to include a page with all the brain state error codes, what they mean, how to handle them?

**Reply by Navid200** [reply]
Thanks for that.  I had left out scanning errors.  Just added it.  It will take the crawlers a little while to propagate.
If someone searches for xDrip G7 troubleshooting, they will find it and everything they need to address any possible issue should be there.  If not, we should add it.  But, going into too much detail would just clutter the page.
